phylogenetic/bin/newreference.py

```diff
@@ -0,0 +1,53 @@
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+from Bio.SeqFeature import SeqFeature, FeatureLocation, Seq
+import shutil
+import argparse
+import sys
+
+def new_reference(referencefile, outgenbank, outfasta, gene):
+    ref = SeqIO.read(referencefile, "genbank")
+    startofgene = None
+    endofgene = None
+    for feature in ref.features:
+        if feature.type == 'source':
+            ref_source_feature = feature
+        if feature.type =='gene' or feature.type == 'CDS':
+            a = list(feature.qualifiers.items())[0][-1][0]
+            if a == gene:
+                startofgene = int(list(feature.location)[0])
+                endofgene =  int(list(feature.location)[-1])+1
+
+    # If user provides a --gene 'some name' is not found, print a warning and use the entire genome.
```

Comment:
```suggestion
    # If user provides a --gene 'some name' that is not found, print a warning and use the entire genome.
```

Reply:
Also, the comment says "print a warning and use the entire genome" but what the code seems to do is print an error message and exit.

Reply:
Agree I'll change it to "print an error instead of silently using entire genome". Gah, forgot to fix docs to refect this during

* https://github.com/nextstrain/rsv/commit/8cd6a1384301161117252f8e6f612f7332bd9416

Will submit a fix to RSV as well

* https://github.com/nextstrain/rsv/blob/a1788ce2c9c4375fb5a06d1426c64c45cf90225f/scripts/newreference.py#L21

Reply:
Just documenting that will be fixed in RSV (https://github.com/nextstrain/rsv/pull/60) and fixed here in https://github.com/nextstrain/dengue/pull/47/commits/dd45b63e22ea40083428d5cdfc1204f2c95c28a6